docs/components/concepts/document-handling.md

```diff
@@ -5,56 +5,85 @@ description: "Learn more about integration, secure management, and efficient sto
 keywords: ["document handling"]
 ---
 
-Offering a more comprehensive approach to document handling, Camunda now provides efficient storage and retrieval of documents across both development and production environments.
+Offering more robust document handling capabilities within Camunda SaaS and Self-Managed, users can now efficiently manage large volumes of binary data such as PDFs and images across both development and production environments. This assists with streamlining business processes involving document uploads, storage, and management, enhancing workflow efficiency and compliance without compromising the performance or integrity of active workflows.
 
-Whether you are working across Connectors, Modeler, or Tasklist, document handling may be beneficial for users who want the following:
+Across the unified Camunda 8 platform, use document handling to store, track, and manage documents utilizing the [Camunda 8 API](/apis-tools/camunda-api-rest/specifications/create-documents.api.mdx), Connectors, Forms, and Tasklist.
 
-- Use the default cloud storage to support, track, and manage documents.
-- [AWS S3](https://aws.amazon.com/s3/) storage and bucket creation per cluster to securely store and retrieve documents in an external, scalable storage solution, and to ensure storage is properly isolated and managed for each environment.
-- Support for the local file system as a storage option for development and testing purposes to set up and manage documents in local environments.
+## Storage integration
 
-## Modeler
+### SaaS
 
-A [form](/components/modeler/forms/camunda-forms-reference.md) can display documents with the [document preview component](/components/modeler/forms/form-element-library/forms-element-library-document-preview.md):
+A [Google Cloud Platform](https://cloud.google.com/storage) storage integration is configured for SaaS and handled by Camunda.
 
-![document preview for form](./assets/document-preview.png)
+#### Limitations
+
+- One bucket per cluster is permitted with SaaS.
+- This storage integration is handled and configured by Camunda. While this is not dynamically configurable by the cluster, it is provided as environment configuration.
+- **Maximum file size**: 10 MB per file.
+- **File expiration time/time-to-live (TTL) policy**: 30 days by default. Clients for Connectors and Forms may specify a custom expiration date when uploading documents.
+
+### Self-Managed
+
+<!--- Does this content belong here, or should we link to it from the SM docs? --->
```

Comment:
@conceptualshark WDYT about the document handling content specific to SM here? Would you place it elsewhere?